operators/tokenizer/clip_tokenizer.hpp

```diff
@@ -0,0 +1,27 @@
+#include <list>
+#include "ocos.h"
+#include "ustring.h"
+#include "string_utils.h"
+
+class VocabData;
+
+struct KernelClipBpeTokenizer : BaseKernel {
+  KernelClipBpeTokenizer(const OrtApi& api, const OrtKernelInfo* info);
+  void Compute(OrtKernelContext* context);
+
+ private:
+  std::vector<int64_t> Tokenize(ustring& input, int64_t max_length);
+
+  int64_t padding_length_;
+  std::list<int> byte_list_;
+  std::shared_ptr<VocabData> bbpe_tokenizer_;
```

Comment:
why does it have to be shared_prt, instead of unique_ptr

Reply:
I believe it is because we are using the same implementation of VocabData for gpt2 and clip tokenizers, and the former requires multiple pointers to it. I tried it still however and changed our implementation accordingly but ran into memory errors while building.